da/celestia/celestia.go

```diff
@@ -111,34 +129,56 @@ func (c *DataAvailabilityLayerClient) SubmitBlock(ctx context.Context, block *ty
 }
 
 // CheckBlockAvailability queries DA layer to check data availability of block at given height.
-func (c *DataAvailabilityLayerClient) CheckBlockAvailability(
-	ctx context.Context, dataLayerHeight uint64,
-) da.ResultCheckBlock {
-	shares, err := c.client.NamespacedShares(ctx, c.namespaceID, dataLayerHeight)
-	code := dataRequestErrorToStatus(err)
-	if code != da.StatusSuccess {
+func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
```

Comment:
Why do you need this call btw? Do you want to check the availability yourself? Note that the node performs that automatically and can also do SharesAvailbale check implicitly when you get Blobs for a height.

Reply:
we never used it. i think we can remove it. @tzdybal any objections?

Reply:
Which API calls block on Data availability? @Wondertan

Reply:
Currently none to be very honest https://github.com/celestiaorg/celestia-node/issues/2385

Reply:
For now, you can already assume it does that, as we do sampling via calling SharesAvailable in background.